src/frontend/lib/first.ts

```diff
@@ -0,0 +1,8 @@
+/**
+ * Get the first value from an iterable, or `undefined` if the iterable is empty.
+ *
+ * Like `myArray[0]`, but works for other iterables like `Set`s.
+ */
+export function first<T>(iterable: Iterable<T>): undefined | T {
+  for (const value of iterable) return value;
```

Comment:
I wonder if this is any different (performance, memory) to `iterable.next().value`? Anyway, inconsequential for this code.

Reply:
Did a quick benchmark comparing this to `return iterable[Symbol.iterator]().next().value` and it was ~0.01ns different. My benchmark was _very_ simplistic but I suspect they're comparable.